### ISSUE TITLE
torch.logaddexp produces NaN in second-order AD for finite logits

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#196704]
### 🐛 Describe the bug

`torch.logaddexp` returns `nan` for a nested reverse-mode second derivative, even though every logit is finite and the mathematical second derivative is finite.
Each `logaddexp` term is smooth; its curvature can be written with finite logistic weights. The claim is the nonfinite autodiff result, not agreement with a tiny last-bit value. At `t = -800.0` the forward value is `10003.0` and matches the reference. The second backward pass returns `nan`.
This issue reproduces on CPU, so CUDA availability does not affect the result.
```python
import torch
dtype = torch.float64
t = torch.tensor(-800.0, dtype=dtype, requires_grad=True)
x = torch.stack((t + 2, -2 * t)).reshape(2, 1)
y = torch.stack((t / 2 - 1, 3 - t, 2 * t + 1)).reshape(1, 3)
w = torch.tensor([[1, 2, -1], [3, -2, 4]], dtype=dtype)
out = (torch.logaddexp(x, y) * w).sum()
g, = torch.autograd.grad(out, t, create_graph=True)
actual, = torch.autograd.grad(g, t)
print("actual:")
print(actual)
print("expected:")
print("finite")
```
### Actual vs expected result
```text
actual:
nan
expected:
finite
```
The discrepancy occurs in the nested (second-order) backward pass. The forward value is `10003.0`. The second derivative should stay finite for these finite logits; PyTorch returns `nan`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A
OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64
Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01
CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128
Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```
